lib/util.js

```diff
@@ -1,35 +1,48 @@
-const CID = require('cids')
+// { cid, binary }
+function isBlock (block) {
+  return typeof block === 'object' && isCID(block.cid) && isBuffer(block.binary)
+}
+
+function isCID (key) {
```

Comment:
Shouldn't the `isCID()` check from `CID` just pass with old and new CIDs?

Reply:
no, I think it uses a symbol check still

So in this case, the latest iteration on my computer even drops the `codec` check so it's basically checking for new CIDs only and old CIDs won't pass. So I should probably drop this function entirely and use `multiformats.CID.isCID()` instead. In most places it's used it falls back to attempting a `new multiformats.CID(thing.toString())` so that should be fine for converting old CIDs.

Trying to mix the two just doesn't work, it's like playing wac-a-mole with errors. I'm discovering that you need to be fully in either ecosystem for any of this to work. Code my my computer has dropped ipld-dag-cbor for the multiformats branch of @ipld/dag-cbor which seems to be working well but it's still using ipld-dag-pb and that gets very awkward as you should be able to see in fixture-data.js. If I was using more complex linking within those PB blocks then I'd be in real trouble but the simplicity of my graphs is saving me.

Things to either fix or document very well.

Reply:
> no, I think it uses a symbol check still

Yes. @mikeal's  plan was to use the same symbol for the new version. I also don't know if that's a good idea.